src/ic-cdk/src/api/call.rs

```diff
@@ -153,6 +153,8 @@ unsafe extern "C" fn callback<T: AsRef<[u8]>>(state_ptr: *const RwLock<CallFutur
             // borrow_mut() the state as well. So we need to be careful to not double-borrow_mut.
             waker.wake()
         }
+    } else {
+        crate::trap("Call already trapped");
```

Comment:
```suggestion
        crate::trap("This call has not been awaited, e.g., because a previous callback trapped");
```
This could be more precise given an offline discussion with @adamspofford-dfinity